cmd/apiserver/app/server/server.go

```diff
@@ -107,9 +107,9 @@ func NewCommandServer(out io.Writer) *cobra.Command {
 	return cmd
 }
 
-// runServer is a method on the options for composition. allows embedding in a
+// RunServer is a method on the options for composition. Allows embedding in a
 // higher level options as we do the etcd and serving options.
-func (serverOptions ServiceCatalogServerOptions) runServer() error {
+func (serverOptions ServiceCatalogServerOptions) RunServer(stopCh <-chan struct{}) error {
```

Comment:
Yes. I am not aware of any upstream code that uses this same setup for anything but tests. We could make a signal handler that acts on SIGwhatever to kill appropriately in the main program. I don't think it would be worth it, as the apiserver has no state, so it can pretty much be killed recklessly.